### PR TITLE
Compute constants

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libevmasm/ConstantOptimiser.cpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		eth::Instruction::MLOAD, // back up memory
+		u256(32),
+		AssemblyItem(PushData, u256(1) << 16), // has to be replaced
+		eth::Instruction::DUP4,
+		eth::Instruction::CODECOPY,
+		eth::Instruction::DUP2,
+		eth::Instruction::MLOAD,
+		eth::Instruction::SWAP2,
+		eth::Instruction::MSTORE
+	}
```
Comment (arkpar) [inline libevmasm/ConstantOptimiser.cpp]: Does not compile with msvc

**Comment by chriseth** [comment]
Fixed (hopefully).

**Comment by chriseth** [comment]
Failures seem to be test infrastructure issues.

File: libevmasm/ConstantOptimiser.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	m_copyRoutine[4] = _assembly.newData(data);
+	replaceConstants(_items, m_copyRoutine);
+}
+
+AssemblyItems ComputeMethod::findRepresentation(u256 const& _value)
+{
+	if (_value < 0x10000)
+		// Very small value, not worth computing
+		return AssemblyItems{_value};
+	else if (dev::bytesRequired(~_value) < dev::bytesRequired(_value))
```
Comment (LianaHus) [inline libevmasm/ConstantOptimiser.cpp]: you could remove dev:: here

Comment (chriseth) [inline libevmasm/ConstantOptimiser.cpp]: Actually not, because there is a method called `bytesRequired`. Removed some other namespace prefixes, though.

**Pull request merged**